### PR TITLE
Integrate RDP algorithm into chart filtering (#321)

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Solve Issue #321

A simple demo to test things out

```
import pandas as pd
import numpy as np
from taipy.gui import Gui

df = pd.DataFrame(np.random.randint(0, 10000000, size=(3000000, 1)), columns=["A"])
df['index1'] = df.index

chart_data = df.head(10)

def get_all_data(state):
    state.chart_data = df

def reset_data(state):
    state.chart_data = df.head(0)

Gui(page="""
<|Get All Data|button|on_action=get_all_data|>

<|Reset|button|on_action=reset_data|>

<|{chart_data}|chart|x=index1|y=A|limit_rows|limit_threshold=300|>
""").run(run_browser=False)

```

### CODE REVIEW

File: gui/src/components/Taipy/Chart.tsx [hunk 1]
```diff
@@ -122,6 +123,7 @@ const Chart = (props: ChartProp) => {
         tp_onRangeChange,
         propagate = true,
         limitRows = false,
+        limitThreshold = undefined,
```
Comment (FredLL-Avaiga) [inline gui/src/components/Taipy/Chart.tsx]: `= undefined` is not necessary

File: taipy/gui/data/pandas_data_accessor.py [hunk 2]
```diff
@@ -238,10 +239,12 @@ def __get_data(  # noqa: C901
             nb_rows_max = payload.get("width")
             # view with the requested columns
             if nb_rows_max and nb_rows_max < len(value) / 2:
-                value = value.copy()
-                value["tp_index"] = value.index
-                # we need to be more clever than this :-)
-                value = value.iloc[:: (len(value) // nb_rows_max)]
+                x_column, y_column = columns[1] if len(columns) > 1 else None, columns[0]
+                try:
+                    threshold = payload.get("limitThreshold", 50)
```
Comment (FredLL-Avaiga) [inline taipy/gui/data/pandas_data_accessor.py]: the 50 value could be defined in the config (give a global default value)

**Review by FredLL-Avaiga — APPROVED** [review]
looks good to me

File: gui/src/components/Taipy/Chart.tsx (hunk 1 above)
Comment (dinhlongviolin1) [reply gui/src/components/Taipy/Chart.tsx]: Thanks!

File: taipy/gui/data/pandas_data_accessor.py (hunk 2 above)
Comment (dinhlongviolin1) [reply taipy/gui/data/pandas_data_accessor.py]: You are absolutely right!

**Review by FabienLelaquais — APPROVED** [review]
👌

**Pull request merged**